public/app/features/query/components/QueryEditorRow.tsx

```diff
@@ -423,7 +423,14 @@ export class QueryEditorRow<TQuery extends DataQuery> extends PureComponent<Prop
     const colour = type === 'warning' ? 'orange' : 'blue';
     const iconName = type === 'warning' ? 'exclamation-triangle' : 'file-landscape-alt';
 
-    const serializedWarnings = uniqueWarnings.map((warning) => warning.text).join('\n');
+    let serializedWarnings, hasMultipleWarnings;
+    if (uniqueWarnings.length > 1) {
+      serializedWarnings = '<ul>' + uniqueWarnings.map((warning) => '<li>' + warning.text + '</li>').join('') + '</ul>';
+      hasMultipleWarnings = true;
+    } else {
+      serializedWarnings = uniqueWarnings[0].text;
+      hasMultipleWarnings = false;
+    }
```

Comment:
this is not really how you do it in react world (it's a last resort), here you can simply pass a ReactNode instead of a string, this is something Tooltip already supports

Reply:
Thanks for the tip! Revised to pass a ReactNode here without changing the Tooltip code.